docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
+
+The commands introduced in this feature include : `add`, `edit`, `list`, `delete`, `get`. 
+The commands are implemented with HashMap and use NRIC as key and the Patient class as value.
+The patient list feature is facilitated by PatientMap class which implements the following operations: 
+
+    - PatientMap #add(Patient patient) — This command adds the patient object into the patient list using the patient’s nric as key.
+
+    - PatientMap #remove(String nric) — This command removes the patient object from the existing patient list. 
+
+    - PatientMap #hasKey(String nric) — This command checks whether the patient object resides in the existing patient list. 
+    
+    
+** Design Considerations ** 
+
+*Aspect: Data Structure of the Patient List* 
+
+        Alternative 1 (current choice): Hash Map
+          Pros: Allow faster lookup of patients’ information using the unique identifier (nric)
+          Cons: Implementation is harder and may result in bugs if not implemented accurately.
+            
+        Alternative 2: Array List
+          Pros: This would be easier to implement and retrieve the information.
+          Cons: When a patient is deleted, all the patients in the patient list need to be checked. 
+                This would cause the deletion to be very slow when there is a large number of patients in the list.
+
+#### 4.1.1 Add Patient Details
+
+The user is able to add patients into the program to keep track of the patient's details. 
+The command: 
+
+    add patient /ic S7777777Z /n Alice /p 98765432 /d 01/01/2000 /b B+ /a Peanuts
+    
+will add a patient with `NRIC` as S7777777Z with the following attributes: 
+* name: `Alice`
+* phone number: `98765432`
+* date of birth: `01/01/2000`
+* blood type: `B+`
+* allergies: `Peanuts`
+
+**Implementation** 
+
+The activity diagram below summarises the process of executing an `add` command.
+![Add Patient Sequence Diagram](images/AddPatientSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `add patient /ic S7777777Z /n Alice /p 98765432 /d 01/01/2000 /b B+ /a Peanuts`.
+
+2. `HappyPills` calls `Parser#parse()` which then calls `PatientParser#parse()`. 
+
+3. `PatientParser#parse()` will call its own method `PatientParser#parseAddCommand()`.
+
+4. `PatientParser#parseAddCommand()` will first validate the attributes and then create an object `AddPatientCommand` 
+if the attributes are valid. 
+
+5. `HappyPills` then calls `AddPatientCommand.execute()` to execute the command. 
+
+6. In `AddPatientCommand.execute()`, a patient is added and the display message is returned.
+
+#### 4.1.2 Edit Patient Details
+
+The user is able to edit the details of a patient in the list of patients currently in the program. The command: 
+
+    edit patient S7777777Z /a School
+    
+will edit the patient's allergies to `School`. 
+
+**Implementation** 
+
+![Edit Patient Sequence Diagram](images/EditPatientSequenceDiagram.jpg)
```

Comment:
There is a typo here (same as the diagram below: Pateint -> Patient)
![bug5](https://user-images.githubusercontent.com/35903974/77999257-f30f2000-7364-11ea-8418-6297f91d5c8d.png)